JSONObject.java

```diff
@@ -1699,10 +1728,8 @@ public String toString() {
      *             If the object contains an invalid number.
      */
     public String toString(int indentFactor) throws JSONException {
-        StringWriter w = new StringWriter();
-        synchronized (w.getBuffer()) {
-            return this.write(w, indentFactor, 0).toString();
-        }
+        StringBuilder w = new StringBuilder();
```

Comment:
Same change related to https://github.com/stleary/JSON-java/issues/54 as above. Using StringBuilder no longer requires the synchronization.